packages/react-core/src/demos/SearchInput.md

```diff
@@ -0,0 +1,299 @@
+          afterDate.setDate(afterDate.getDate() + 1);
+          toDate.setMonth(toDate.getMonth() + 2);
+          toDate.setDate(toDate.getDate() + 1);
+          break;
+        case '6 months':
+          afterDate.setMonth(afterDate.getMonth() - 6);
+          afterDate.setDate(afterDate.getDate() + 1);
+          toDate.setMonth(toDate.getMonth() + 6);
+          toDate.setDate(toDate.getDate() + 1);
+          break;
+        case '1 year':
+          afterDate.setFullYear(afterDate.getFullYear() - 1);
+          afterDate.setDate(afterDate.getDate() + 1);
+          toDate.setFullYear(toDate.getFullYear() + 1);
+          toDate.setDate(toDate.getDate() + 1);
+          break;
+      }
+      setValue(`${hasWords && (hasWords + " ")}after:${yyyyMMddFormat(afterDate)} to:${yyyyMMddFormat(toDate)}`)
+    } else {
+      setValue(hasWords);
+    }
+    
+    setIsAdvancedSearchOpen(false);
+  };
+  
+  const searchInput = (
+    <SearchInput
+      value={value}
+      onChange={onChange}
+      onToggleAdvancedSearch={(e, isOpen) => {
+        e.stopPropagation();
+        setIsAdvancedSearchOpen(isOpen)
+      }}
+      isAdvancedSearchOpen={isAdvancedSearchOpen}
+      onClear={onClear}
+      onSearch={onSubmit}
+      ref={searchInputRef}
+      id="custom-advanced-search"
+    />
+  );
+  
+  // Clicking the 'date within' toggle should open its associated menu and then place the browser 
+  // focus on the first menu item. 
+  const toggleDateWithinMenu = ev => {
+    ev.stopPropagation(); // Stop handleClickOutside from handling
+    setTimeout(() => {
+      if (dateWithinMenuRef.current) {
+        const firstElement = dateWithinMenuRef.current.querySelector('li > button:not(:disabled)');
+        firstElement && firstElement.focus();
+      }
+    }, 0);
+    setIsDateWithinOpen(!isDateWithinOpen);
+  };
+  
+  const dateWithinOptions = (
+    <Menu ref={dateWithinMenuRef} selected={dateWithin} onSelect={(e, itemId) => {
+      e.stopPropagation();
+      setIsDateWithinOpen(false);
+      setDateWithin(itemId);
+    }}>
+      <MenuContent>
+        <MenuList>
+          <MenuItem itemId="1 day">1 day</MenuItem>
+          <MenuItem itemId="3 days">3 days</MenuItem>
+          <MenuItem itemId="1 week">1 week</MenuItem>
+          <MenuItem itemId="2 weeks">2 weeks</MenuItem>
+          <MenuItem itemId="1 month">1 month</MenuItem>
+          <MenuItem itemId="2 months">2 months</MenuItem>
+          <MenuItem itemId="6 months">6 months</MenuItem>
+          <MenuItem itemId="1 year">1 year</MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
+  
+  const dateWithinToggle = (
+    <MenuToggle ref={dateWithinToggleRef} onClick={toggleDateWithinMenu} isExpanded={isDateWithinOpen} style={{width: "100%"}}>
+      {dateWithin}
+    </MenuToggle>
+  );
+  
+  const advancedForm = (
+    <div ref={advancedSearchPaneRef} role="dialog" aria-label="Advanced search form">
+      <Form>
+        <Card>
```

Comment:
The card component doesn't match the search input styling. Can we expose the `.pf-c-search-input__menu` and `.pf-c-search-input__menu-body` elements and use those instead? https://github.com/patternfly/patternfly-react/blob/06b7d17f1b48996f17fd1021f1834eaeebd7bc18/packages/react-core/src/components/SearchInput/SearchInput.tsx#L358

Reply:
In this case, the menu is not a child of search input - so the styles are not applied correctly to the form if they are in orphaned `.pf-c-search-input__menu` and `.pf-c-search-input__menu-body`components.

Reply:
Hmm... I'm not sure I understand. The rendered markup all seems to support the styling working, I just renamed the card elements to search input menu/menu-body and moved the actions into the form as an action group:

<img width="674" alt="Screen Shot 2021-09-10 at 1 38 04 PM" src="https://user-images.githubusercontent.com/35148959/132901707-47c546d3-fa16-40b8-af56-52a224504a1e.png">